src/main/java/roboy/dialog/states/ordinaryStates/GamingTwentyQuestionsState.java

```diff
@@ -0,0 +1,263 @@
+			return Output.say("Now think of a character and tell me when you're ready to start the game.");
+
+		} else if(guessesAvailable) {
+
+			try {
+				return doAGuess();
+			}
+			catch (Exception e){
+				return Output.say("I have a problem finding a guess...");
+			}
+		}
+		else {
+
+			return askNextQuestion();
+		}
+
+	}
+
+	@Override
+	public Output react(Interpretation input) {
+
+
+		String intent = getIntent (input);
+
+		if(!isUserReady(intent)){
+
+			return Output.say("Nice, then let's start");
+
+		} else if(guessesAvailable){
+
+			return processUserGuessAnswer(intent);
+
+		} else {
+
+			try {
+
+				return saveUsersEstimate(intent);
+			}
+			catch (Exception e){
+				return Output.say("I have a problem processing your input...");
+			}
+		}
+	}
+
+	@Override
+	public State getNextState() {
+
+		if(gameFinished){
+			applyFilter(winner);
+			resetGame();
+			return getTransition(TRANSITION_GAME_ENDED);
+
+		} else {
+			return this;
+		}
+	}
+
+	private String getIntent(Interpretation input) {
+
+
+		List<String	> tokens = input.getTokens();
+
+
+
+
+
+		String intent = null;
+
+		for(String token : tokens) {
+			if(token.equals("yes")) {
+				intent = "yes";
+			} else if (token.equals("no")) {
+				intent = "no";
+			} else if(token.equals("ready")) {
+				intent = "ready";
+			} else if(token.equals("back")) {
+				intent = "back";
+			} else if(token.equals("dont")) {
+				intent = "dont know";
+			} else if(token.equals("not")) {
+				intent = "probably not";
+			} else if(token.equals("probably")) {
+				intent = "probably";
+			}
+		}
+
+		return intent;
+	}
+
+	private Output askNextQuestion() throws NullPointerException{
+
+		String nextQuestionString = null;
+		try {
+			nextQuestion = aw.getCurrentQuestion();
+			nextQuestionString = nextQuestion.getQuestion();
+			++numberGuesses;
+		}
+		catch(NullPointerException e){
+			gameFinished = true;
+			return Output.say("I throw in the towel, I think I have no idea what to ask next. You win!");
+		}
+		return Output.say(nextQuestionString);
+	}
+
+	private Output saveUsersEstimate(String intent) throws Exception {
+
+		//check if guesses above threshold are available
+		if(aw.getGuessesAboveProbability(PROBABILITY_THRESHOLD).size() > 0){
+			guessesAvailable = true;
+		}
+
+		if (intent != null) {
+			switch (intent) {
+				case "yes":
+					aw.answerCurrentQuestion(Answer.YES);
+					return Output.sayNothing();
+				//break;
+				case "no":
+					aw.answerCurrentQuestion(Answer.NO);
+					return Output.sayNothing();
+				//break;
+
+				case "dont know":
+					aw.answerCurrentQuestion(Answer.DONT_KNOW);
+					return Output.sayNothing();
+				//break;
+
+				case "probably":
+					aw.answerCurrentQuestion(Answer.PROBABLY);
+					return Output.sayNothing();
+				//break;
+
+				case "probably not":
+					aw.answerCurrentQuestion(Answer.PROBABLY_NOT);
+					return Output.sayNothing();
+				//break;
+
+				case "back":
+					aw.undoAnswer();
+					return Output.sayNothing();
```

Comment:
in all of this cases we should probably say something to confirm roboy understood it correctly 
so change Output.sayNothing() to a reaction to the intent